src/phoenix/experimental/evals/functions/generate.py

```diff
@@ -54,35 +57,45 @@ def llm_generate(
         should correspond to the column names of the output dataframe. If None, the output dataframe
         will have a single column named "output". Default None.
 
+        concurrency (int, default=20): The number of concurrent evals if async submission is
+        possible.
+
     Returns:
         generations_dataframe (pandas.DataFrame): A dataframe where each row
         represents the generated output
 
     """
     tqdm_bar_format = get_tqdm_progress_bar_formatter("llm_generate")
     output_parser = output_parser or _no_op_parser
-    with set_verbosity(model, verbose) as verbose_model:
-        template = normalize_prompt_template(template)
-        logger.info(f"Template: \n{template.prompt()}\n")
-        logger.info(f"Template variables: {template.variables}")
-        prompts = map_template(dataframe, template)
-
-        # For each prompt, generate and parse the response
-        output = []
-        # Wrap the loop in a try / catch so that we can still return a dataframe
-        # even if the process is interrupted
-        try:
-            for prompt in tqdm(prompts, bar_format=tqdm_bar_format):
-                logger.info(f"Prompt: {prompt}")
-                response = verbose_model(prompt, instruction=system_instruction)
-                parsed_response = output_parser(response)
-                output.append(parsed_response)
-
-        except (Exception, KeyboardInterrupt) as e:
-            logger.error(e)
-            print(
-                "Process was interrupted. The return value will be incomplete",
-                e,
+    template = normalize_prompt_template(template)
+    logger.info(f"Template: \n{template.prompt()}\n")
+    logger.info(f"Template variables: {template.variables}")
+    prompts = map_template(dataframe, template)
+
+    async def _run_llm_generation_async(prompt: str) -> Any:
+        with set_verbosity(model, verbose) as verbose_model:
+            response = await verbose_model._async_generate(
+                prompt,
+                instruction=system_instruction,
+            )
+        return output_parser(response)
+
+    def _run_llm_generation_sync(prompt: str) -> Any:
```

Comment:
```suggestion
    async def _run_llm_generation_async(prompt: str) -> Dict[str, Any]:
        with set_verbosity(model, verbose) as verbose_model:
            response = await verbose_model._async_generate(
                prompt,
                instruction=system_instruction,
            )
        return output_parser(response)

    def _run_llm_generation_sync(prompt: str) -> Dict[str, Any]:
```

Suggestion: We can be more specific here.